GOVERNANCE.md

```diff
@@ -0,0 +1,458 @@
+
+Some of its goals are:
+
+- Making sure that the different stakeholders are represented with their requirements
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues in their assigned area, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the 'top' are the maintainers, whose job is to define and realize the project vision of thin-edge.io.
+Maintainers should strive for agreement based on the following factors:
+
+- The vision of the whole project
+- Feasibility, accounting for future technical and cultural shifts
+- The health of the project going forward
+
+Overall, the maintainers should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the maintainers are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try to make sure to explore possible options together before calling for time-consuming solutions like a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If an exclusive choice has to be made (where it is not possible to entertain two conflicting approaches), and no clear side is more advantageous to pick, a vote should be held with the majority opinion being the final decision.
+
+- Some decisions are local, e.g. a plugin does not impact others with its behaviour, however it still needs to be a 'good citizen' in the project (CI, format, etc..)
+    - These should be clearly scoped in their respective project part
+    - However, if needed a 'higher up' decision can be requested if no consensus is achieved
+
+
+
+## Project Structure
+
+The project has a hierarchy to solve these two problems:
+
+1. How can users make sure that their voice and interests are represented in the project?
```

Comment:
I would answer this: thin-edge.io is community driven, representing the interests and implementing the feedback of users who leverage thin-edge.io for their IoT projects is our highest priority. Therefore the maintainers also ensure that users and heard and proactively engage with users through our communication channels: github issues/PRs, discussions or our community events, discord (or even occasional 1to1 conversations) to collect as much feedback and patterns. The maintainers are supported by a wider team of thin-edge.io project members like PMs, contributors to collect the feedback and collect evidence of the most important topics to work on next.

Reply:
@TheNeikos  @didier-wenzek after the merge we need to ensure that maintainer role definition contains that "maintainers ensure that users/community feedback is considered)

Reply:
Follow-up task: https://github.com/thin-edge/thin-edge.io/issues/1377